New project: generatePackageJson doesn't include tslib as a dependency

## Current Behavior
When a new workspace is created with a node app (eg. express) and the `generatePackageJson` option is set to true, `package.json` is missing `tslib` which is required since the `tsconfig.base.json` that comes with a new `nx` workspace uses `importHelpers: true`. This means that when pulled into a Docker container for deployment or similar, and `npm install` is run, `tslib` is excluded, and the build will fail.

## Expected Behavior
`tslib` should be included in the `package.json` file produced by the `generatePackageJson` option if `importHelpers: true`.

## Steps to Reproduce
Repo that demonstrates this behaviour: `https://github.com/jksaunders/nx-generate-package-json-tslib`
To use the example, run `npm install` then `nx build`, and observe that the generated `package.json` in the `dist/api` folder is missing `tslib`.

To reproduce from scratch:
- `npx create-nx-workspace --preset=express`
- `nx g @nrwl/express:app api`
- `nx build`
- observe that the generated `package.json` in the `dist/api` folder is missing `tslib`.

### Failure Logs
When `npm ci/install` is run in the container:
```
2022-06-29 11:16:12.392 MDT
Error: Cannot find module 'tslib'

2022-06-29 11:16:12.392 MDT
Require stack:
2022-06-29 11:16:12.392 MDT
- /usr/src/app/main.js
2022-06-29 11:16:12.392 MDT
 at Module._resolveFilename (node:internal/modules/cjs/loader:946:15)
2022-06-29 11:16:12.392 MDT
 at Module._load (node:internal/modules/cjs/loader:787:27)
2022-06-29 11:16:12.392 MDT
 at Module.require (node:internal/modules/cjs/loader:1012:19)
2022-06-29 11:16:12.392 MDT
 at require (node:internal/modules/cjs/helpers:102:18)
2022-06-29 11:16:12.392 MDT
 at Object.tslib (/usr/src/app/main.js:22:18)

2022-06-29 11:16:12.392 MDT
 at __webpack_require__ (/usr/src/app/main.js:46:41)
2022-06-29 11:16:12.392 MDT
 at /usr/src/app/main.js:59:17
2022-06-29 11:16:12.392 MDT
 at /usr/src/app/main.js:98:3
2022-06-29 11:16:12.392 MDT
 at Object.<anonymous> (/usr/src/app/main.js:103:12)
2022-06-29 11:16:12.392 MDT
 at Module._compile (node:internal/modules/cjs/loader:1112:14) {
2022-06-29 11:16:12.392 MDT
 code: 'MODULE_NOT_FOUND',
2022-06-29 11:16:12.392 MDT
 requireStack: [ '/usr/src/app/main.js' ]
2022-06-29 11:16:12.392 MDT
}
```

### Environment
```
 >  NX   Report complete - copy this into the issue template

   Node : 18.3.0
   OS   : darwin x64
   npm  : 8.11.0
   
   nx : 14.3.6
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.3.6
   @nrwl/eslint-plugin-nx : 14.3.6
   @nrwl/express : 14.3.6
   @nrwl/jest : 14.3.6
   @nrwl/js : 14.3.6
   @nrwl/linter : 14.3.6
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : 14.3.6
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/workspace : 14.3.6
   typescript : 4.7.4
   ---------------------------------------
   Community plugins:
```

--- a/e2e/node/src/node.test.ts
+++ b/e2e/node/src/node.test.ts
@@ -279,6 +279,7 @@ describe('Build Node apps', () => {
           '@nestjs/platform-express': '^9.0.0',
           'reflect-metadata': '^0.1.13',
           rxjs: '^7.0.0',
+          tslib: '^2.3.0',
         },
         main: 'main.js',
         name: expect.any(String),

--- a/packages/express/src/generators/init/init.spec.ts
+++ b/packages/express/src/generators/init/init.spec.ts
@@ -27,6 +27,8 @@ describe('init', () => {
     const packageJson = readJson(tree, 'package.json');
     // add express
     expect(packageJson.dependencies['express']).toBeDefined();
+    // add tslib
+    expect(packageJson.dependencies['tslib']).toBeDefined();
     // move `@nrwl/express` to dev
     expect(packageJson.dependencies['@nrwl/express']).toBeUndefined();
     expect(packageJson.devDependencies['@nrwl/express']).toBeDefined();

--- a/packages/express/src/generators/init/init.ts
+++ b/packages/express/src/generators/init/init.ts
@@ -6,6 +6,7 @@ import {
   Tree,
 } from '@nrwl/devkit';
 import { initGenerator as nodeInitGenerator } from '@nrwl/node';
+import { tslibVersion } from '@nrwl/node/src/utils/versions';
 import {
   expressTypingsVersion,
   expressVersion,
@@ -20,6 +21,7 @@ function updateDependencies(tree: Tree) {
     tree,
     {
       express: expressVersion,
+      tslib: tslibVersion,
     },
     {
       '@types/express': expressTypingsVersion,

--- a/packages/node/src/utils/generate-package-json-webpack-plugin.ts
+++ b/packages/node/src/utils/generate-package-json-webpack-plugin.ts
@@ -7,7 +7,11 @@ import {
 import { createPackageJson } from '@nrwl/workspace/src/utilities/create-package-json';
 
 import type { BuildNodeBuilderOptions } from './types';
-import { getHelperDependenciesFromProjectGraph } from '@nrwl/js/src/utils/compiler-helper-dependency';
+import {
+  getHelperDependenciesFromProjectGraph,
+  HelperDependency,
+} from '@nrwl/js/src/utils/compiler-helper-dependency';
+import { readTsConfig } from '@nrwl/workspace/src/utilities/typescript';
 
 export class GeneratePackageJsonWebpackPlugin implements WebpackPluginInstance {
   constructor(
@@ -30,6 +34,22 @@ export class GeneratePackageJsonWebpackPlugin implements WebpackPluginInstance {
             this.context.root,
             this.context.projectName
           );
+
+          const importHelpers = !!readTsConfig(this.options.tsConfig).options
+            .importHelpers;
+          const shouldAddHelperDependency =
+            importHelpers &&
+            helperDependencies.every(
+              (dep) => dep.target !== HelperDependency.tsc
+            );
+
+          if (shouldAddHelperDependency) {
+            helperDependencies.push({
+              type: 'static',
+              source: this.context.projectName,
+              target: HelperDependency.tsc,
+            });
+          }
 
           if (helperDependencies.length > 0) {
             this.projectGraph.dependencies[this.context.projectName] =

